Core/Core/People/ContextEnrollment.swift

```diff
@@ -0,0 +1,194 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2023-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Foundation
+import CoreData
+
+final public class ContextEnrollment: NSManagedObject {
```

Comment:
The main reason to have this entity is to be able to display the user's role within the course so I'd remove any other property not directly supporting this. The copy pasted grade formatting is super error prone for the future, even this copy pasted logic is outdated since we have LGO related code in the original implementation already which is missing from here.

Reply:
cleaned up.